src/ucp/core/ucp_types.h

```diff
@@ -61,6 +61,31 @@ typedef struct ucp_rkey_config_key      ucp_rkey_config_key_t;
 typedef struct ucp_proto                ucp_proto_t;
 
 
+/**
+ * UCP TL bitmap
+ *
+ * Bitmap type for representing which TL resources are in use.
+ */
+typedef ucs_bitmap_t(128) ucp_tl_bitmap_t;
```

Comment:
UCP_MAX_RESOURCES

Reply:
Can't do that, because the `UCS_PP_TOKENPASTE` macro can't be used for the struct type definition.
`typedef struct { ....} UCS_PP_TOKENPASTE(....);` won't compile.

Reply:
```
#include <ucs/sys/preprocessor.h>
...
#define ucs_bitmap_t(_length) UCS_PP_TOKENPASTE3(ucs_bitmap_, _length, _t)
```